lib/src/grpc/mod.rs

```diff
@@ -0,0 +1,477 @@
+    }
+    fn get_inner(&self) -> &Arc<Mutex<Option<T>>> {
+        &self.inner
+    }
+}
+
+/// Returns a ([String], [i32]) host, port from provided environment variables.
+/// Returns default values (localhost, 50051) if environment variable is not found.
+pub fn get_endpoint_from_env(env_host: &str, env_port: &str) -> (String, u16) {
+    grpc_debug!("(get_endpoint_from_env) entry");
+
+    let host = match std::env::var(env_host) {
+        Ok(val) => val,
+        Err(_) => {
+            grpc_error!(
+                "(get_endpoint_from_env) {} undefined, using default [localhost].",
+                env_host
+            );
+            "localhost".to_string()
+        }
+    };
+    let port: u16 = match std::env::var(env_port) {
+        Ok(val) => match val.parse::<u16>() {
+            Ok(val) => val,
+            Err(_) => {
+                grpc_error!(
+                    "(get_endpoint_from_env) {} is not a valid u16 type, using default [50051].",
+                    env_port
+                );
+                50051
+            }
+        },
+        Err(_) => {
+            grpc_error!(
+                "(get_endpoint_from_env) {} undefined, using default [50051].",
+                env_port
+            );
+            50051
+        }
+    };
+
+    grpc_info!("(get_endpoint_from_env) host [{}], port [{}].", host, port);
+    (host, port)
+}
+
+#[macro_export]
+/// Implements [`ClientConnect`] trait for provided gRPC Client
+macro_rules! grpc_client {
+    ($rpc_service_client: ident) => {
+        #[async_trait]
+        impl $crate::grpc::ClientConnect<$rpc_service_client<Channel>>
+            for $crate::grpc::GrpcClient<$rpc_service_client<Channel>>
+        {
+            /// Get a connected client object
+            async fn connect(
+                &self,
+            ) -> Result<$rpc_service_client<Channel>, tonic::transport::Error> {
+                $rpc_service_client::connect(self.get_address()).await
+            }
+        }
+    };
+}
+
+#[cfg(test)]
+mod tests {
+    use super::mock::*;
+    use super::*;
+    use tonic::{transport::Channel, Request, Response};
+
+    pub mod grpc_client {
+        #![allow(unused_qualifications)]
+        include!("mock/grpc_client.rs");
+    }
+    use grpc_client::{client::MockClient, ReadyRequest};
+    pub mod grpc_server {
+        #![allow(unused_qualifications)]
+        include!("mock/grpc_server.rs");
+    }
+    use grpc_server::server::{MockServer, MockService};
+    grpc_client!(MockClient);
+
+    /// Mock struct to implement our MockService for success tests
+    #[derive(Default, Debug, Clone, Copy)]
+    pub struct GrpcMockSuccess {}
+
+    #[tonic::async_trait]
+    impl MockService for GrpcMockSuccess {
+        async fn is_ready(
+            &self,
+            request: Request<grpc_server::ReadyRequest>,
+        ) -> Result<Response<grpc_server::ReadyResponse>, Status> {
+            println!("Got a request: {:?}", request);
```

Comment:
prepend with "(is_ready MOCK)" here and L241

Reply:
https://github.com/Arrow-air/lib-common/pull/12/commits/2e66ec06357b84e3c7a2e9612408b6e3f3f39e9c